templates/create-daml-app-test-resources/index.test.ts

```diff
@@ -118,26 +96,19 @@ beforeAll(async () => {
   // Ensure the UI server is ready by checking that the port is available.
   await waitOn({resources: [`tcp:localhost:${UI_PORT}`]});
 
-  // Launch a browser once for all tests.
+  // Launch a single browser for all tests.
   browser = await puppeteer.launch();
 }, 40_000);
 
 afterAll(async () => {
-  // Kill the sandbox and JSON API processes.
-  // Note that `kill()` sends the `SIGTERM` signal but the actual processes may
-  // not die immediately.
+  // Kill the `daml start` process, allowing the sandbox and JSON API server to
+  // shut down gracefully.
+  // The latter process should also remove the JSON API port file.
   // TODO: Test this on Windows.
-  if (sandbox) {
-    sandbox.kill();
-  }
-  if (jsonApiServer) {
-    jsonApiServer.kill();
+  if (startProc) {
+    startProc.kill('SIGTERM');
   }
 
-  // The sandbox does not clean up its port file on termination so we take care
-  // of that. (The JSON API server takes care of this for us.)
-  await removeFile(SANDBOX_PORT_FILE_PATH);
```

Comment:
Shouldn't we now remove the port file for the JSON API? Maybe doing it immediately after we finished waiting for it is the right moment.

Reply:
The JSON API server should remove it if it's shut down gracefully (which it should be at this point in the code). I added the removal at the start in case it was previously killed abruptly.

Reply:
Ah, I didn't know the JSON API server will remove the file. That's great. Thanks for pointing that out!